### PR TITLE
Summary

### PR DESCRIPTION
**Pull request by Wikitelmex**
- follow the layout of the wireframes provided. personalized the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
    -  built with Devise.
- Food list:
    - display a list of food as in the wireframe.
    - lead to the form that allows users to add new food.
- Recipies list:
    - display a list of recipes created by the logged-in user as in the wireframe.
    - lead to recipe details.
    - If the user is the owner of the recipe, should allow user to delete it.
- Public recipe list:
    - display a list of all public recipes ordered by newest as in the wireframe.
    - lead to recipe details.
- Recipe details:
    - If the recipe is public or the user is the owner of the recipe, display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, lead to the form that allows user to add new food.

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+  # before_action :configure_sign_up_params, only: [:create]
+  # before_action :configure_account_update_params, only: [:update]
+
+  # GET /resource/sign_up
+  # def new
+  #   super
+  # end
```
Comment (t-yanick) [inline app/controllers/users/registrations_controller.rb]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  # The path used after sign up.
+  # def after_sign_up_path_for(resource)
+  #   super(resource)
+  # end
+
+  # The path used after sign up for inactive accounts.
+  # def after_inactive_sign_up_path_for(resource)
+  #   super(resource)
+  # end
```
Comment (t-yanick) [inline app/controllers/users/registrations_controller.rb]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

File: app/controllers/users/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+class Users::SessionsController < Devise::SessionsController
+  # before_action :configure_sign_in_params, only: [:create]
+
+  # GET /resource/sign_in
+  # def new
+  #   super
+  # end
+
+  # POST /resource/sign_in
```
Comment (t-yanick) [inline app/controllers/users/sessions_controller.rb]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

File: app/controllers/users/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    sign_out(current_user)
+    redirect_to root_path
+  end
+
+  # protected
+
+  # If you have extra params to permit, append them to the sanitizer.
+  # def configure_sign_in_params
+  #   devise_parameter_sanitizer.permit(:sign_in, keys: [:attribute])
+  # end
```
Comment (t-yanick) [inline app/controllers/users/sessions_controller.rb]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (t-yanick) [inline app/models/user.rb]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

File: db/migrate/20220207193241_devise_create_users.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
```
Comment (t-yanick) [inline db/migrate/20220207193241_devise_create_users.rb]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

File: spec/helpers/home_helper_spec.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+# Specs in this file have access to a helper object that includes
+# the HomeHelper. For example:
+#
+# describe HomeHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
```
Comment (t-yanick) [inline spec/helpers/home_helper_spec.rb]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

File: spec/helpers/ingredients_helper_spec.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+# Specs in this file have access to a helper object that includes
+# the IngredientsHelper. For example:
+#
+# describe IngredientsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
```
Comment (t-yanick) [inline spec/helpers/ingredients_helper_spec.rb]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

File: spec/helpers/public_recipes_helper_spec.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+# describe PublicRecipesHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe PublicRecipesHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (t-yanick) [inline spec/helpers/public_recipes_helper_spec.rb:5]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

File: spec/helpers/recipes_helper_spec.rb [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+# describe RecipesHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe RecipesHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (t-yanick) [inline spec/helpers/recipes_helper_spec.rb:5]: - [x] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/users/sessions_controller.rb (hunk 3 above)
Comment (Wikitelmex) [reply app/controllers/users/sessions_controller.rb]: done

File: app/controllers/users/registrations_controller.rb (hunk 2 above)
Comment (Wikitelmex) [reply app/controllers/users/registrations_controller.rb]: done

**Review by t-yanick — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**